core/src/testUtils/groovy/nu/yona/server/test/AppService.groovy

```diff
@@ -0,0 +1,406 @@
+package nu.yona.server.test
+
+import groovy.json.*
+import groovyx.net.http.RESTClient
+import groovyx.net.http.URIBuilder
+
+import java.text.SimpleDateFormat
+
+import nu.yona.server.YonaServer
+
+class AppService extends Service
+{
+	final GOALS_PATH = "/goals/"
+	final USERS_PATH = "/users/"
+	final BUDDIES_PATH_FRAGMENT = "/buddies/"
+	final DIRECT_MESSAGES_PATH_FRAGMENT = "/messages/direct/"
+	final ANONYMOUS_MESSAGES_PATH_FRAGMENT = "/messages/anonymous/"
+	final ALL_MESSAGES_PATH_FRAGMENT = "/messages/all/"
+	final RELEVANT_CATEGORIES_PATH_FRAGMENT = "/relevantCategories/"
+	final NEW_DEVICE_REQUEST_PATH_FRAGMENT = "/newDeviceRequest"
+	final MOBILE_NUMBER_CONFIRMATION_PATH_FRAGMENT = "/confirmMobileNumber"
+
+	JsonSlurper jsonSlurper = new JsonSlurper()
+
+	AppService ()
+	{
+		super("yona.appservice.url", "http://localhost:8082")
+	}
+
+	void confirmMobileNumber(Closure asserter, User user)
+	{
+		def response = confirmMobileNumber(user.mobileNumberConfirmationUrl, """{ "code":"${user.mobileNumberConfirmationCode}" }""", user.password)
+		asserter(response)
+	}
+
+	def addUser(Closure asserter, password, firstName, lastName, nickname, mobileNumber, devices, goals, parameters = [:])
+	{
+		def jsonStr = User.makeUserJsonString(firstName, lastName, nickname, mobileNumber, devices, goals, [])
+		def response = addUser(jsonStr, password, parameters)
+		asserter(response)
+		return (isSuccess(response)) ? new User(response.responseData, password) : null
+	}
+
+	def addUser(jsonString, password, parameters = [:])
+	{
+		yonaServer.createResourceWithPassword(USERS_PATH, jsonString, password, parameters)
+	}
+
+	def assertUserCreationResponseDetails(def response)
+	{
+		assertUserOverwriteResponseDetails(response)
+		assert response.responseData.confirmationCode
+	}
+
+	def assertUserUpdateResponseDetails(def response)
+	{
+		assert response.status == 200
+		assertUserWithPrivateData(response.responseData)
+	}
+
+	def assertUserOverwriteResponseDetails(def response)
+	{
+		assertResponseStatusCreated(response)
+		assertUserWithPrivateData(response.responseData)
+	}
+
+	def assertUserGetResponseDetailsWithPrivateDataCreatedOnBoddyRequest(def response)
```

Comment:
Then you should rename it to `assertValidVPNProfile` and have a method for the user created on buddy request in the `CreateUserOnBuddyRequestTest`.